sensor/common/detector/baseline/baseline.go

```diff
@@ -75,16 +89,199 @@ func (w *baselineEvaluator) AddBaseline(baseline *storage.ProcessBaseline) {
 	}
 	containerNameMap[baseline.GetKey().GetContainerName()] = baselineSet
 
-	log.Infof("Successfully added process baseline %s", baseline.GetId())
+	log.Debugf("Successfully added process baseline %s", baseline.GetId())
 }
 
 // IsInBaseline checks if the process indicator is within a locked baseline
 // If the baseline does not exist, then we return true
 func (w *baselineEvaluator) IsOutsideLockedBaseline(pi *storage.ProcessIndicator) bool {
+	if pi == nil {
+		return false // Treat nil process as within baseline
+	}
+
 	w.baselineLock.RLock()
 	defer w.baselineLock.RUnlock()
 
 	baseline := w.baselines[pi.GetDeploymentId()][pi.GetContainerName()]
 	// If there is no baseline, then we are counting it as if it's within the baseline
 	return baseline != nil && !baseline.Contains(processbaseline.BaselineItemFromProcess(pi))
 }
+
+// optimizedBaselineEvaluator implements memory-optimized baseline evaluation using process set deduplication
+type optimizedBaselineEvaluator struct {
```

Comment:
nit: How about placing the entire code of `optimizedBaselineEvaluator` (all lines below this comment) in a separate file?

Reply:
Done in https://github.com/stackrox/stackrox/pull/17019